File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        left = 0
        right = 0
        zeroes = 0
        maxLen = 0

        while (right < len(nums)):
            if (nums[right] == 0):
                zeroes += 1
            if (zeroes > k):
                while (nums[left] != 0):
                    left += 1
                zeroes -= 1
                left += 1
            else:
                length = right - left + 1
                maxLen = max(maxLen, length)
            right += 1
        return maxLen
        

File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int currZeroes = 0;
        int maxLength = 0;
        int left = 0;
        for (int right = 0; right< nums.length; right++)    {
            if (nums[right] == 0)   {
                while (left <= right && currZeroes == k) {
                    if (nums[left] == 0)    {
                        currZeroes--;
                    }
                    left++;
                }
                currZeroes++;
            }
            int length = right - left + 1;
            maxLength = Math.max(maxLength, length);
        }
        return maxLength;
    }
}